src/ports/postgres/modules/cols_vec/cols2vec.py_in

```diff
@@ -0,0 +1,110 @@
+    """
+    Args:
+        @param schema_madlib:               Name of MADlib schema
+        @param model:                       Name of table containing the tree model
+        @param source_table:                Name of table containing prediction data
+        @param output_table:                Name of table to output the results
+        @param list_of_features:            Comma-separated string of column names or
+                                            expressions to put into feature array.
+                                            Can also be a '*' implying all columns
+                                            are to be put into feature array.
+        @param list_of_features_to_exclude: Comma-separated string of column names
+                                            to exclude from the feature array
+        @param cols_to_output:              Comma-separated string of column names
+                                            from the source table to keep in the output table,
+                                            in addition to the feature array.
+
+    Returns:
+        None
+
+    """
+
+    with MinWarning('warning'):
+        validate_cols2vec_args(source_table, output_table, list_of_features,
+                               list_of_features_to_exclude, cols_to_output, **kwargs)
+
+        all_cols = ''
+        feature_cols = ''
+        if list_of_features.strip() == '*':
+            all_cols = get_cols(source_table, schema_madlib)
+            all_col_set = set(list(all_cols))
+            exclude_set = set(split_quoted_delimited_str(
+                list_of_features_to_exclude))
+            feature_set = all_col_set - exclude_set
+            feature_cols = py_list_to_sql_string(
+                list(feature_set), "text", False)
+            filtered_list_of_features = ",".join(
+                feat for feat in list(feature_set))
+        else:
+            feature_list = split_quoted_delimited_str(list_of_features)
+            feature_exclude = split_quoted_delimited_str(
+                list_of_features_to_exclude)
+            return_set = set(feature_list) - set(feature_exclude)
+            feature_cols = py_list_to_sql_string(
+                list(return_set), "text", False)
+            filtered_list_of_features = ",".join(
+                feat for feat in feature_list if feat in return_set)
+
+        output_cols = ''
+        if cols_to_output:
+            output_cols_list = [', '.join(get_cols(source_table, schema_madlib)) if col == '*' else col
+                                for col in split_quoted_delimited_str(cols_to_output)]
+            output_cols = ', '.join(output_cols_list) + ","
+
+        plpy.execute("""
+    		CREATE TABLE {output_table} AS
+    		select {output_cols}
+            array[{filtered_list_of_features}] as feature_vector
+    		from {source_table}
+    		""".format(**locals()))
+
+        plpy.execute("""
```

Comment:
I haven't understood the need for the summary table. Is it just to record the features combined in the array? Can we provide that as an output of the function? Creating a table just to record that one parameter seems unnecessary.

Reply:
@iyerr3  The vec2cols story (https://issues.apache.org/jira/browse/MADLIB-1240) might consume this summary table if provided as the `dictionary` param in that module.

Reply:
If there are 1000+ columns which you want to keep track of, then saving the array or col names in a summary tables might be convenient.   It is not ideal but should not be onerous to the user and they can ignore the summary table if they don't care about it.